packages/application-generic/src/encryption/encrypt-provider.ts

```diff
@@ -41,19 +44,34 @@ export function decryptCredentials(
 
   for (const key in credentials) {
     decryptedCredentials[key] =
-      typeof credentials[key] === 'string' &&
-      isEncryptedCredential(credentials[key])
-        ? decryptProviderSecret(credentials[key])
+      typeof credentials[key] === 'string' && novuEncrypted(credentials[key])
+        ? decryptSecret(credentials[key])
         : credentials[key];
   }
 
   return decryptedCredentials;
 }
 
-function isEncryptedCredential(text: string): boolean {
+export function encryptApiKey(apiKey: string): EncryptedSecret {
+  if (novuEncrypted(apiKey)) {
+    return apiKey;
+  }
+
+  return encryptSecret(apiKey);
+}
+
+export function decryptApiKey(apiKey: string): string {
+  if (novuEncrypted(apiKey)) {
+    return decryptSecret(apiKey);
+  }
+
+  return apiKey;
+}
+
+function novuEncrypted(text: string): text is EncryptedSecret {
```

Comment:
Prefixing with `is` to provide better description that return value is boolean.

```suggestion
function isNovuEncrypted(text: string): text is EncryptedSecret {
```

Reply:
Love it, I totally agree, before novu I always prefixed with 'is' but in novu it felt to me that people liked it less.